pkg/forwarder/blocked_endpoints.go

```diff
@@ -29,32 +36,38 @@ func newBlockedEndpoints() *blockedEndpoints {
 	return &blockedEndpoints{errorPerEndpoint: make(map[string]*block)}
 }
 
-func (e *blockedEndpoints) block(endpointName string) {
+func (e *blockedEndpoints) close(endpoint string) {
 	e.m.Lock()
 	defer e.m.Unlock()
 
 	var b *block
-	if knownBlock, ok := e.errorPerEndpoint[endpointName]; ok {
+	if knownBlock, ok := e.errorPerEndpoint[endpoint]; ok {
 		b = knownBlock
 	} else {
 		b = &block{}
 	}
-	b.nbError++
 
-	newInterval := time.Duration(b.nbError) * blockInterval
-	if newInterval > maxRetryInterval {
-		newInterval = maxBlockInterval
-	}
-	b.until = time.Now().Add(newInterval)
+	b.nbError = int(math.Min(float64(maxErrors), float64(b.nbError+1)))
```

Comment:
sometimes instead of doing a lot of casting you can just:
```go
b.nbError += 1
if b.nbError > maxErrors {
  b.nbError = maxErrors
}
```
makes it way more readable ;)

Reply:
Thanks so much!